spiffworkflow-backend/src/spiffworkflow_backend/services/process_instance_processor.py

```diff
@@ -1810,9 +1812,20 @@ def suspend(self) -> None:
         )
         db.session.commit()
 
+    def bring_archived_future_tasks_back_to_life(self) -> None:
+        archived_future_tasks = (
+            self.process_instance_model.future_tasks_query()
+            .filter(FutureTaskModel.archived_for_process_instance_status == True)  # noqa: E712
+            .all()
+        )
+        for archived_future_task in archived_future_tasks:
+            archived_future_task.archived_for_process_instance_status = False
+            db.session.add(archived_future_task)
+
     def resume(self) -> None:
         self.process_instance_model.status = ProcessInstanceStatus.waiting.value
         db.session.add(self.process_instance_model)
+        self.bring_archived_future_tasks_back_to_life()
```

Comment:
since it is committing at the end of the resume method, it should be logically equivalent to what you're suggesting. it will only commit if all operations succeed.